modules/apps/remote-app/remote-app-admin-web/src/main/java/com/liferay/remote/app/admin/web/internal/frontend/taglib/clay/data/set/CustomElementPortletClayDataSetEntry.java

```diff
@@ -38,6 +39,10 @@ public String getName() {
 		return _customElementPortletEntry.getName(_locale);
 	}
 
+	public String getPortletName() {
```

Comment:
The portlet name is really ugly (vanilla_counter_40149) how is it generated?

Reply:
It is the name of the custom element plus the id of the `CustomElementPortletEntry`. Ray's PR had the concept of `portlet alias` which has to be unique and is used for this, however I don't like it too much, because it means addding a new validated field to the registation, which is just used for routing, thus its use will be quite unclear for the rest of people not needing to use it.

However, I don't like this automatic names either, but preferred it to push it like these to make this conversation happen and decide what to do with this.

Some other options may be:

1. Generate a unique random word combination (similar to what https://www.npmjs.com/ does in the top left header, and some other cloud services do when you deploy/configure something).
2. Add a portlet name field and prefill it with the current algorithm by default, then let the user modify it if needed.
3. Generate a synthetic portlet name based on its human readable description (and make that description unique)
4. ...

Reply:
Ok, I see the code under `CustomElementPortlet`.

1. is very alien in a Liferay environment... I would go with 2. (friendly URL style). I don't see how 3. can scale with a very long description.